app/actions/apps.ts

```diff
@@ -63,9 +63,12 @@ export function doAppCall<Res=unknown>(call: AppCall, intl: any): ActionFunc {
         } catch (err) {
             const msg = err.message || 'We found an unexpected error.';
             ephemeral(msg);
+
             return {
-                type: AppCallResponseTypes.ERROR,
-                error: msg,
+                data: {
+                    type: AppCallResponseTypes.ERROR,
+                    error: {message: msg},
+                },
```

Comment:
@enahum @mickmister @larkox As I understand the issue here - we optimized for the simplicity for (Lambda) App developers. An App responds to Calls with a singular `AppCallResponse` struct, which can represent all possible responses. `[AppCallResponse]Type` is an enum of `ok`, `form`, `error`, [etc...]. At some point we had a separate `AppCallResponse.Error|Data|Form` fields that were used for the respective response types, but then we decided that the pair of `Type|Data` was sufficient.

JSON returned by the Apps reflects ^^, and the code here merely handles it.

Reply:
I can understand that. In my case because I have no idea of that, it clearly confuses me as everything else in this repo is structured with a different format, in that case, to solve it, the action should handle the response and return the values following the format of the actions where data contains relevant data from the response and and error property if it applies.

thoughts?